code/src/pgsql_io/pgsql_io.ml

```diff
@@ -961,6 +1116,36 @@ and create_sm_login ?passwd ~user t =
   let open Abbs_future_combinators.Infix_result_monad in
   Io.wait_for_frames t >>= fun frames -> create_sm_process_login_frames ?passwd ~user t frames
 
+(* TODO: Maybe it's worth aglutinating step_02 and step_03
+   under the same recursive function, left it for a future
+   refactor *)
+and create_sm_scram_sha256_step_03 ?passwd ~user client_final t =
+  let open Pgsql_codec.Frame.Backend in
```

Comment:
Same comment as I add below (I actually reviewed this in reverse order for some reason).